src/components/Misc/DeleteOrResetModal.tsx

```diff
@@ -0,0 +1,130 @@
+import { Box, Button, Modal, Stack, Text, TextInput } from '@mantine/core';
+import classes from './styles/DeleteOrResetModal.module.css';
+import { ChangeEvent, useCallback, useState } from 'react';
+
+type BaseProps = {
+	isOpen: boolean;
+	onClose: () => void;
+	header: string;
+	specialContent?: React.ReactNode;
+	content: string;
+	actionProcessingContent?: React.ReactNode;
+	isActionInProgress?: boolean;
+	onConfirm: () => void;
+};
+
+// Note: The `confirmationText` and `placeholder` props are required for 'delete' and 'reset' types, but not for 'simple' type.
+type DeleteOrResetModalProps =
+	| (BaseProps & {
+			type: 'simple';
```

Comment:
Is there a default type or every time we have to pass a type? We can have a default type.

Reply:
The `type` prop is a must-have in the current setup. It makes sure we’re doing things explicitly. I intentionally added it to avoid the hassle of remembering the default type from the developer.

Reply:
Makes sense